File: frontend/src/interfaces/companyInfo.ts
import type { FormValues } from "../lib/form";

export interface CompanyInfoData {
	id: number;
	companyName: string;
	address: string;
	phone: string;
	email: string;
	website?: string;
	logoUrl?: string;
	taxNumber?: string;
	registrationNumber?: string;
}

export interface CompanyInfoFormData extends FormValues {
	companyName: string;
	address: string;
	phone: string;
	email: string;
	website: string;
	logoUrl: string;
	taxNumber: string;
	registrationNumber: string;
}

File: backend/src/main/java/com/hpms/backend/model/CompanyInfo.java
package com.hpms.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "company_info")
public class CompanyInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String companyName;

    @Column(nullable = false, length = 500)
    private String address;

    @Column(nullable = false, length = 50)
    private String phone;

    @Column(nullable = false, length = 255)
    private String email;

    @Column(length = 255)
    private String website;

    @Column(length = 500)
    private String logoUrl;

    @Column(length = 100)
    private String taxNumber;

    @Column(length = 100)
    private String registrationNumber;
}